app/controllers/stories_controller.rb

```diff
@@ -438,13 +438,19 @@ def disown
       return render plain: "can't find story", status: 400
     end
 
-    InactiveUser.disown! story
-    @story = find_story
-    @comments = Comment.story_threads(@story).for_presentation
+    if request.get?
+      InactiveUser.disown! story
```

Comment:
This introduces a CSRF so any page on the web could include `<img src="https://lobste.rs/s/abc123/disown">` so that when the submitter of story `abc123` loads the page, their browser sends a GET request and the story is disowned without their knowledge or action.

Also, it is a bad Rails practice to check the HTTP verb inside an action. The route should recognize the http verb and send to two different actions.